core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -378,15 +374,20 @@ class JsonProtocolSuite extends SparkFunSuite {
     val oldInfo = JsonProtocol.accumulableInfoFromJson(oldJson)
     assert(false === oldInfo.internal)
   }
+}
 
-  /** -------------------------- *
-   | Helper test running methods |
-   * --------------------------- */
 
-  private def testEvent(event: SparkListenerEvent, jsonString: String) {
+// This extends SparkFunSuite only because we want its `assert` method.
+private[spark] object JsonProtocolSuite extends SparkFunSuite {
```

Comment:
What's the motivation for moving these into a separate object? Are you trying to reuse some of these methods in a different suite?